src/cargo/core/compiler/unit_graph.rs

```diff
@@ -127,9 +126,9 @@ pub fn emit_serialized_unit_graph(
         roots,
     };
 
-    let stdout = std::io::stdout();
-    let mut lock = stdout.lock();
-    serde_json::to_writer(&mut lock, &s)?;
-    drop(writeln!(lock));
+    let mut shell = gctx.shell();
+    let stdout = shell.out();
+    serde_json::to_writer(&mut *stdout, &s)?;
+    writeln!(stdout)?;
```

Comment:
Maybe better using `gctx.shell().print_json(&result)?` as we have that function

Reply:
Thanks! applied :)